packages/widget/src/controllers/wallet-manager/manager.ts

```diff
@@ -110,12 +124,27 @@ export class WalletController implements ReactiveController {
     }
   };
 
-  switchChain(chainId: number): void {
+  async switchEvmChain(
+    chainId: number,
+    provider: Eip1193Provider
+  ): Promise<void> {
     if (this.walletContext.value?.evmWallet) {
-      void this.walletContext.value.evmWallet.provider.request({
-        method: 'wallet_switchEthereumChain',
-        params: [{ chainId: utils.hexValue(chainId) }]
-      });
+      try {
+        await provider.request({
+          method: 'wallet_switchEthereumChain',
+          params: [{ chainId: utils.hexValue(chainId) }]
+        });
+      } catch (switchError) {
+        const chainData = (await (
```

Comment:
caching the response will save redundant network calls

Reply:
I think the effort here to add logic to cache the response could be out of the scope of the issue, considering that this block is only going to be triggered if the user switch to a network that he doesn't have on his wallet IMHO